syntax = "proto3";

package scale_predictor;

message PredictRequest {
    string function_name = 1;
    repeated int32 window = 2;
    int32 index = 3;
}

message PredictResponse {
    int32 result = 1;
}

service ScalePredictor {
    rpc Predict (PredictRequest) returns (PredictResponse);
}